pyfstat/core.py

```diff
@@ -40,6 +40,12 @@
 class BaseSearchClass(object):
     """ The base search class providing parent methods to other searches """
 
+    def __new__(cls, *args, **kwargs):
+        logging.info(f"Creating {cls.__name__} object...")
+        instance = super().__new__(cls)
```

Comment:
I'm new to `__new__`, but shouldn't this pass on `, *args, **kwargs` here...?

Reply:
My understanding is "No" since `__new__` falls back to the most fundamental class (which I believe is `object`) which takes nothing. 

The reason for those `*args, **kwargs` is because child classes will try to pass further parameters to `__new__` (the same ones as `__init__` I believe) but we don't care about them since we are only interested in adding a default print every time a new object is instantiated.

Reply:
[Let me quote the actual docs](https://docs.python.org/3/reference/datamodel.html#object.__new__)

- `__new__` gets the same arguments as `__init__`.
- After returning a new instance, `__init__` is run on that instance using the arguments passed to `__new__`.

In other words, whatever is going into `__init__` goes first through `__new__`. Since we don't do anything fancy in `__new__`, we can ignore all of those parameters and fallback to the default `__new__`  method; but we still need to allow for such parameters to get included as they will then go to `__init__`, where we do need them.

Reply:
"using super().__new__(cls[, ...]) with appropriate arguments" seems to imply the (kw)args should be passed through, doesn't it?

Reply:
But the examples I found recommending args/kwargs passing actually all seem to be from python2 days, and don't run in python 3.7.3 themselves. So: objection retracted; if your version runs, let's go with it.